src/github.com/getlantern/fronted/direct.go

```diff
@@ -36,31 +53,23 @@ func getCertPool() *x509.CertPool {
 	return pool
 }
 
-type Direct struct {
+type direct struct {
 	tlsConfigs      map[string]*tls.Config
 	tlsConfigsMutex sync.Mutex
 }
 
-func NewDirect() *Direct {
-	return &Direct{
+func NewDirect() *direct {
+	d := &direct{
 		tlsConfigs: make(map[string]*tls.Config),
 	}
+	d.fillMasquerades()
+	return d
 }
 
-func (d *Direct) getMasquerade() *Masquerade {
-	if len(masqCh) > 0 {
-		return <-masqCh
-	}
-	log.Debugf("Refreshing live masquerades")
-	ms := <-masqueradesCh
-	size := len(ms)
-	for i := 0; i < 10; i++ {
+func (d *direct) fillMasquerades() {
+	for i := 0; i < 40; i++ {
```

Comment:
We only want to dial first 40 masquerades from shuffled sets in parallel and ignore the rest?

I guess we can just spawn a few goroutines, say, 10, and loop in the goroutine when error encountered to dial one masquerade, so we always have and only have a few valid masquerades in stock. Testing against one masquerade will fail soon unless times out, so it will not slow down the caller of `getMasquerade` I think.